### PR TITLE
Use bulk_update for updating blob download counts

### PR DESCRIPTION
**Pull request by danlamanna**
This removes a linear scaling set of update queries in favor of `bulk_update`. This should prevent us from hitting our preconfigured time limits for processing S3 log files.

### CODE REVIEW

**Review by mvandenburgh — APPROVED** [review]

File: dandiapi/analytics/tasks/__init__.py [hunk 1]
```diff
@@ -80,10 +81,20 @@ def process_s3_log_file_task(s3_log_key: str) -> None:
                 logger.info('Already processed log file %s', s3_log_key)
             return
 
+        # we need to store all of the fully hydrated blob objects in memory in order to use
+        # bulk_update, but this turns out to not be very costly. 1,000 blobs use about ~8kb
+        # of memory.
+        # breakpoint()
```
Comment (jjnesbitt) [inline dandiapi/analytics/tasks/__init__.py:87]:
```suggestion
```

Comment (danlamanna) [reply dandiapi/analytics/tasks/__init__.py:87]: :upside_down_face:

File: dandiapi/analytics/tasks/__init__.py [hunk 2]
```diff
@@ -80,10 +81,19 @@ def process_s3_log_file_task(s3_log_key: str) -> None:
...
+        # bulk_update, but this turns out to not be very costly. 1,000 blobs use about ~8kb
+        # of memory.
+        asset_blobs = []
+
+        # batch the blob queries to avoid a large WHERE IN clause
+        for batch in batched(download_counts, 1_000):
+            asset_blobs += AssetBlob.objects.filter(blob__in=batch)
+
+        for asset_blob in asset_blobs:
+            asset_blob.download_count = F('download_count') + download_counts[asset_blob.blob]
```
Comment (jjnesbitt) [inline dandiapi/analytics/tasks/__init__.py:94]: What does using `F('download_count')` do here? As opposed to just writing:

```python
asset_blob.download_count += download_counts[asset_blob.blob]
```

Comment (danlamanna) [reply dandiapi/analytics/tasks/__init__.py:94]:
```sql
-- assuming download_count is already 20 and we're adding 10

-- using F('download_count')
UPDATE asset_blob SET download_count = download_count + 10;

-- without
UPDATE asset_blob SET download_count = 30;
```

The latter is susceptible to the lost update problem.

**Comment by yarikoptic** [comment]
PLEASE review/chime in on 
- https://github.com/dandi/dandi-archive/issues/1900

We will also meet with @CodyCBakerPhD  this Wed 10-11am to go through what he has done in https://github.com/dandi/dandiarchive-logs/issues/10#issuecomment-2339068431 etc

**Comment by danlamanna** [comment]
> PLEASE review/chime in on
> 
>     * [Should process_s3_log_file_task be removed/turned off? #1900](https://github.com/dandi/dandi-archive/issues/1900)

I'll leave a comment there, but I think this is still worth deploying regardless.

**Pull request merged**